### PR TITLE
Fix up many linting issues

### PR DESCRIPTION
**Pull request by rtpg**
Most of these are fixing up issues with the type declarations in customization variables. These all were highlighted by `eask`.

There's also a handful of other fixups:
- pass in the port when creating a service process 
- replace `(eq foo [])` with `(equal foo [])` in a couple of spots (first one does an identity check so would never pass)

### CODE REVIEW

File: clients/lsp-camel.el [hunk 1]
```diff
@@ -20,9 +20,8 @@
 ;; Directory in which the servers will be installed. Lsp Server Install Dir: ~/.emacs.d/.cache/camells
 (defcustom lsp-camel-jar-file (f-join lsp-server-install-dir "camells" lsp-camel-jar-name)
   "Camel Language server jar command."
-  :type 'string
   :group 'lsp-camel
-  :type 'file
+  :type 'directory
```
Comment (kiennq) [inline clients/lsp-camel.el:24]: this should be `file` instead of `directory`, no?

Comment (rtpg) [reply clients/lsp-camel.el:24]: Ah, got confused from the comment above and seeing the word `dir` too many times. Definitely should be `file`

File: clients/lsp-kotlin.el [hunk 2]
```diff
@@ -50,7 +50,10 @@ executable with `exec-path'."
 
 (defcustom lsp-kotlin-trace-server "off"
   "Traces the communication between VSCode and the Kotlin language server."
-  :type '(choice (:tag "off" "messages" "verbose"))
+  :type '(choice (const "off")
+                 (const "messages")
+                 (const "verbose")
+         )
```
Comment (jcs090218) [inline clients/lsp-kotlin.el:56]: Can you move `)` to the previous line? Thanks!

File: clients/lsp-markdown.el [hunk 3]
```diff
@@ -59,9 +59,9 @@ For a complete list of plugins, check:
 
 For a complete list of plugins, check:
  https://github.com/unifiedjs/unified-language-server/blob/main/CONFIGURATION.md#re-using-settings"
-  :type '(choice (
+  :type '(choice
                   (const "retext-english")
-                  (const "remark-parse")))
+                  (const "remark-parse"))
```
Comment (jcs090218) [inline clients/lsp-markdown.el:64]: Can you fix the style to:

```elisp
  :type '(choice (const "retext-english")
                 (const "remark-parse")))
                 (const "remark-parse"))
```

File: clients/lsp-lua.el [hunk 4]
```diff
@@ -383,7 +391,13 @@ and the language server will provide special support.
...
 (defcustom lsp-lua-runtime-version "Lua 5.4"
   "Lua runtime version."
-  :type '(choice (:tag "Lua 5.1" "Lua 5.2" "Lua 5.3" "Lua 5.4" "LuaJIT"))
+  :type '(choice
+                  (const "Lua 5.1")
+                  (const "Lua 5.2")
+                  (const "Lua 5.3")
+                  (const "Lua 5.4")
+                  (const "LuaJIT")
+                  )
```
Comment (jcs090218) [inline clients/lsp-lua.el:400]: Can you fix the style to:

```elisp
  :type '(choice (const "Lua 5.1")
                 (const "Lua 5.2")
                 (const "Lua 5.3")
                 (const "Lua 5.4")
                 (const "LuaJIT"))
```

File: clients/lsp-perlnavigator.el [hunk 5]
```diff
@@ -63,33 +63,40 @@ default to ~/.perlcriticrc. (no aliases, .bat files or ~/)."
   :group 'lsp-perlnavigator
   :package-version '(lsp-mode . "9.0.0"))
 
+(defconst lsp-perlnavigator--log-level-type
+  '(choice  (const "error")
+    (const "warning")
+    (const "info")
+    (const "hint")
+    (const "none")
+    ))
```
Comment (jcs090218) [inline clients/lsp-perlnavigator.el:72]: Can you fix the style to:

```elisp
'(choice  (const "error")
          (const "warning")
          (const "info")
          (const "hint")
          (const "none")))
```

And add a simple docstring to the variable? :)

File: clients/lsp-yaml.el [hunk 6]
```diff
@@ -172,14 +172,21 @@ Limited for performance reasons."
                                         (lsp-package-ensure 'yaml-language-server
                                                             callback error-callback))))
 
+(defconst lsp--yaml-schema-extension-type
+      '(list
+       (cons 'name  string)
+       (cons 'description string)
+       (cons 'url string)
+       (cons 'fileMatch (repeat string))))
```
Comment (jcs090218) [inline clients/lsp-yaml.el:180]: Can you add documentation to this variable? The indentation seems a bit off here. 🤔

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
Thank you!

**Comment by rtpg** [comment]
Sorry about not getting back to this in a timely manner! End of year stuff. Thanks for getting this over the finish line

**Pull request merged**